history-deleter/history.js

```diff
@@ -0,0 +1,58 @@
+// A useful way to extract the domain from a url.
+function get_hostname(url) {
+  var a = document.createElement('a');
+  a.href = url;
+  return a.hostname;
+}
+
+function no_history(hostname) {
+  document.getElementById('history').innerHTML = `No history for ${hostname}.`;
```

Comment:
This is kind of duplicating what's on line 15. Instead of this you could simply inline this in the callback and set the `innerHTML` of `list` in there. Of course, this is a demo for teaching purposes, so if you feel it's clearer to leave this as a separate function that's valid too.